toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -195,6 +227,14 @@ func buildSRPMFile(agent buildagents.BuildAgent, buildAttempts int, srpmFile, ou
 	logBaseName := filepath.Base(srpmFile) + ".log"
 	err = retry.Run(func() (buildErr error) {
 		builtFiles, logFile, buildErr = agent.BuildPackage(srpmFile, logBaseName, outArch, dependencies)
+		// If the package builds with no errors and RUN_CHECK=y, check logs to see if the %check section passed, and if not, return as the build error.
+		if buildErr != nil {
+			return
+		}
+
+		if agent.Config().RunCheck {
+			buildErr = parseCheckSection(logFile)
```

Comment:
Ah, one more thing I just realized: we don't want to fail the whole build if we find a test failure. At least not yet. Once all of our package tests start passing reliably, we can enable that. We might want to also consider adding a build variable like `IGNORE_TEST_FAILURES`, but that's out of scope of this PR.
```suggestion
			parseCheckSection(logfile)
```